packages/docusaurus-plugin-content-blog/src/__tests__/index.test.ts

```diff
@@ -414,14 +414,17 @@ describe('loadBlog', () => {
     );
     const blogPosts = await getBlogPosts(siteDir);
     const noDateSource = path.posix.join('@site', PluginPath, 'no date.md');
-    const noDateSourceBirthTime = (
-      await fs.stat(noDateSource.replace('@site', siteDir))
-    ).birthtime;
+    const noDateSourceFile = path.posix.join(siteDir, PluginPath, 'no date.md');
+    // we know the file exist and we know we have git
+    const result = shell.exec(
+      `git log --follow --max-count=1 --diff-filter=A --format=%ct -- "${noDateSourceFile}"`,
+    );
+    const noDateSourceTime = new Date(Number(result.stdout.trim()) * 1000);
```

Comment:
This was what I initially thought as well:

https://github.com/facebook/docusaurus/pull/6593/commits/8266faeea3d16dfdf29201e3dcba45c59e01e08c#diff-26647d6e7772a1e7730ab6c042efeb7de88b80318e622821e59c802e4af33dd9R417

But it turns out that the _committer_ date inferred by git may differ depending on how much commits were fetched during clone (`--depth`/_shallow clone_).

I do agree with you that it would be better not to use git though. I thought on maybe mocking `shell.exec` to always return a fixed date, but that isn't as good as well.

Do you have any other ideas?

Reply:
I see

So our CI is currently not passing with a hardcoded static date?

The good news is that if you extract this as a function, you can reuse the function in the test 😄 still better than duplicating the git fn

Reply:
> So our CI is currently not passing with a hardcoded static date?

Nope: https://github.com/facebook/docusaurus/runs/5047157194?check_suite_focus=true#step:5:6823

> The good news is that if you extract this as a function, you can reuse the function in the test 😄 still better than duplicating the git fn

Yes. I'll do it.